### DISCUSSION TITLE
Still building?

### DISCUSSION THREAD

**Discussion by sunsetlover36** [passes-org/passes#8] [General]
GM. I want to play with Passes. It's an interesting approach to replace some old/legacy auth methods. Is repo still somehow active and working? Are docs relevant as well, or you're building new thing now from this?

**Comment by danscan** [comment]
gm! The docs are up-to-date, the protocol is stable, and everything ought to work – but let me know if you run into any issues! Happy to help if so.

What are you building?

**Reply by sunsetlover36** [reply]
For now - nothing, but I'll definitely try Passes. Just want to play with basics at first, then maybe some ideas will visit my head. Thank you for the answer!

**Reply by danscan** [reply]
Sounds good. I happen to be writing a separate but related protocol right now for establishing RPC sessions between apps, which I'll publish soon.

Think WalletConnect, but:
- It's at least 10x simpler. No websockets, all HTTP
- It works reliably and SDKs are easy af to build
- It's agnostic to the type of RPC, so not limited to crypto use cases. And for crypto, it can support any RPC, so EVM, Solana, Bitcoin and on
- And it doesn't need a central party to host a relay. RPC sessions can be established between arbitrary apps

If any of that is interesting, definitely lmk :)

**Reply by sunsetlover36** [reply]
This is very interesting. Let's be in touch.

**Reply by sunsetlover36** [reply]
Hey @danscan

Could you share Passes example for React app? I saw you wrote your example using Vue in docs, but code snippet is not visible for me. For example, signed EVM JSON-RPC transaction or maybe something smaller. It would be great, because I can't do much with docs now.

When I'm trying to send request, app fails:
```js
@passes_polyfill.js?v=bc5157d8:46 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'addEventListener')
    at Object.polyfillRequest [as request] (@passes_polyfill.js?v=bc5157d8:46:24)
    at _RequestTopic.sendRequest (@passes_reqs.js?v=bc5157d8:286:35)
```

There was another error, but I fixed it with importing `@passes/polyfill` into my React code. Maybe I'm missing Pass Provider? How can I register one then?

**Comment by danscan** [comment]
Hey, I’m away from my computer until tomorrow, but from what I can tell it
looks like the Pass Provider window failed to open.

Did that error occur in the browser? The passes polyfill only works on the
client, as a heads-up.

I’m wondering if the separate RPC peering protocol I mentioned I’m working
on might be a better fit for what you’re looking for. I expect to publish
it in the next week, and it will be easier to get started with than Passes
is currently!

I’ll also take a look at the issue here tomorrow if it’s still occurring
for you – let me know if the issue is occurring in the browser and I’ll
investigate!

**Reply by sunsetlover36** [reply]
Yes, this error occurred in browser (vite react app)

Thanks!

**Reply by sunsetlover36** [reply]
I managed to resolve it and finally fully understand the concept. I really like Passes' approach to managing user data between apps—it's like a breath of fresh air. I'll think about what I can build with this tool to explore it further. I'm already considering cross-app interaction opportunities where one app could gather data from various topic providers and be built on top of this combined data. Or, as the docs mentioned, there could be some Pass Providers with user's AI models where user can allow apps to utilize their AI models. Honestly, that sounds incredibly exciting.

**Comment by danscan** [comment]
I love to hear it! Really exciting to see someone help themselves to it and
get the concept.

I agree. One way or another I think this pattern is really important –
imagine: apps you use can discover each other and interoperate without
either of their developers even knowing the other app exists.

Passes is one of three protocols I’m working on in this category. The
landscape as I see it:

- Passes: a way for apps to discover each other and communicate and make
requests to the end user without the end user needing any kind of special
setup or browser extension

- RPC peering: a similar setup to Passes with a different UX. An initiating
app can request to connect to a responding app, as long they support each
other’s protocol requirements. The user has to initiate the connection by
opening a link from the initiating app in the responding app (probably
easiest via QR for now), and then the apps can communicate directly going
forward

- Network extensions: the “backend” version of the above two protocols.
Basically, imagine if all your devices were on a VPN where an API was
available at a certain standard host name. Apps running on any of your
devices could automatically discover and request to connect to an API that
could serve as your means of authentication as well as a relay for your
apps to communicate

**Reply by sunsetlover36** [reply]
Wow. Amazing! Looking forward to see new updates.

**Reply by danscan** [reply]
Just dropped a very early preview: https://github.com/danscan/rpcpeering

Bear with me – there's no docs site yet. I'm working on it :)

**Reply by sunsetlover36** [reply]
Will look into it!
